### PR TITLE
[MIG][16.0] delivery_procurement_group_carrier

### PR DESCRIPTION
**Pull request by rousseldenis**
I'm not sure all code should remain in this module (sale action_draft() and carrier_id on procurement_group), so, I'm not sure about module name.

### CODE REVIEW

File: delivery_procurement_group_carrier/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2020 Camptocamp (https://www.camptocamp.com)
+# Copyright 2020-2021 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (lmignon) [inline delivery_procurement_group_carrier/models/sale_order_line.py]: @rousseldenis `s/http/https`

File: delivery_procurement_group_carrier/tests/test_carrier.py [hunk 2]
```diff
@@ -1,45 +1,48 @@
...
-        so = self.env["sale.order"].create(sale_order_vals)
-        so.action_confirm()
-        self.assertTrue(so.picking_ids)
-        self.assertEqual(so.procurement_group_id.carrier_id, so.carrier_id)
+
+        with Form(self.env["sale.order"]) as sale_form:
+            sale_form.partner_id = self.partner
+            with sale_form.order_line.new() as line_form:
+                line_form.product_id = product
+        sale = sale_form.save()
```
Comment (lmignon) [inline delivery_procurement_group_carrier/tests/test_carrier.py]: @rousseldenis You should no more use `Form` to create a SO. Odoo core has been modified to use computed non readonly fields for required fields.

**Review by lmignon — CHANGES_REQUESTED** [review]

File: delivery_procurement_group_carrier/models/sale_order_line.py (hunk 1 above)
Comment (rousseldenis) [reply delivery_procurement_group_carrier/models/sale_order_line.py]: @lmignon Done

**Review by lmignon — APPROVED** [review]
LGTM (Code + functionnal)

**Comment by rousseldenis** [comment]
@jbaudoux

**Review by petrus-v — APPROVED** [review]
LGTM (code review)

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
> I'm not sure all code should remain in this module (sale action_draft() and carrier_id on procurement_group), so, I'm not sure about module name.

When the SO is put back to draft, you can change the carrier. That's why the procurement group is removed as it could be invalid after confirm.

**Review by sebalix — APPROVED** [review]

**Pull request merged**